classes/class-wc-connect-paypal-ec.php

```diff
@@ -0,0 +1,250 @@
+			}
+			return $body;
+		}
+
+		/**
+		 * Get WCS PayPal proxy endpoint
+		 */
+		public function request_endpoint( $endpoint, $environment ) {
+			return trailingslashit( WOOCOMMERCE_CONNECT_SERVER_URL ) . 'paypal/nvp/' . $environment;
+		}
+
+		/**
+		 * Limit supported payment gateway features to payments
+		 */
+		public function supports( $supported, $feature, $gateway ) {
+			return 'ppec_paypal' === $gateway->id ? 'products' === $feature : $supported;
+		}
+
+		public function maybe_set_banner( $order_id ) {
+			$order          = wc_get_order( $order_id );
+			$payment_method = WC_Connect_Compatibility::instance()->get_payment_method( $order );
+
+			if ( 'ppec_paypal' === $payment_method ) {
+				update_option( 'wc_connect_banner_ppec', 'yes' );
+			}
+		}
+
+		/**
+		 * Once a payment is received, show prompt to connect a PayPal account on certain screens
+		 */
+		public function maybe_init_banner() {
+			if (
+				'yes' !== get_option( 'wc_connect_banner_ppec', null ) ||
+				'yes' === get_option( 'wc_connect_dismiss_banner_ppec', null )
+			) {
+				return;
+			}
+
+			$screen = get_current_screen();
+			if ( // Display if on any of these admin pages.
+				( // Orders list.
+					'shop_order' === $screen->post_type
+					&& 'edit' === $screen->base
+					)
+				|| ( // Edit order page.
+					'shop_order' === $screen->post_type
+					&& 'post' === $screen->base
+					)
+				|| ( // WooCommerce settings.
+					'woocommerce_page_wc-settings' === $screen->base
+					)
+				|| ( // WooCommerce featured extension page
+					'woocommerce_page_wc-addons' === $screen->base
+					&& isset( $_GET['section'] ) && 'featured' === $_GET['section']
+					)
+				|| ( // WooCommerce payment gateway extension page
+					'woocommerce_page_wc-addons' === $screen->base
+					&& isset( $_GET['section'] ) && 'payment_gateways' === $_GET['section']
+					)
+				|| 'plugins' === $screen->base
+			) {
+				wp_enqueue_style( 'wc_connect_banner' );
+				add_action( 'admin_notices', array( $this, 'banner' ) );
+			}
+		}
+
+		public function banner() {
+			$prompt = __( 'Link a new or existing PayPal account to enable PayPal Express Checkout features beyond simply taking payments: issue refunds, capture charges after order completion, and more.', 'woocommerce-services' );
+
+			$this->nux->show_nux_banner( array(
+				'title'          => __( 'Connect a PayPal account', 'woocommerce-services' ),
+				'description'    => esc_html( $prompt ),
+				'button_text'    => __( 'Connect', 'woocommerce-services' ),
+				'button_link'    => wc_gateway_ppec()->ips->get_signup_url( 'live' ),
+				'image_url'      => plugins_url( 'images/cashier.svg', dirname( __FILE__ ) ),
+				'should_show_jp' => false,
+				'dismiss_option' => 'ppec',
+			) );
+		}
+
+		public function initialize_settings() {
+			$settings = get_option( 'woocommerce_ppec_paypal_settings', array() );
+
+			if ( ! isset( $settings['button_size'] ) ) {
```

Comment:
Please add a comment explaining why this is a good setting to use for this purpose.